OpenRA.Game/ObjectCreator.cs

```diff
@@ -41,24 +42,47 @@ public ObjectCreator(Manifest manifest, InstalledMods mods)
 				var resolvedPath = FileSystem.FileSystem.ResolveAssemblyPath(path, manifest, mods);
 				if (resolvedPath == null)
 					throw new FileNotFoundException("Assembly `{0}` not found.".F(path));
+#if !MONO
+				var loader = new AssemblyLoader(resolvedPath);
+				var platformType = loader.LoadDefaultAssembly();
+				assemblyList.Add(platformType);
+
+#else
+				LoadAssembly(assemblyList, resolvedPath);
+#endif
+			}
+
+			AppDomain.CurrentDomain.AssemblyResolve += ResolveAssembly;
+			assemblies = assemblyList.SelectMany(asm => asm.GetNamespaces().Select(ns => (asm, ns))).ToArray();
+		}
 
-				// .NET doesn't provide any way of querying the metadata of an assembly without either:
-				//   (a) loading duplicate data into the application domain, breaking the world.
-				//   (b) crashing if the assembly has already been loaded.
-				// We can't check the internal name of the assembly, so we'll work off the data instead
-				var hash = CryptoUtil.SHA1Hash(File.ReadAllBytes(resolvedPath));
+		void LoadAssembly(List<Assembly> assemblyList, string resolvedPath)
+		{
+			// .NET doesn't provide any way of querying the metadata of an assembly without either:
+			//   (a) loading duplicate data into the application domain, breaking the world.
+			//   (b) crashing if the assembly has already been loaded.
+			// We can't check the internal name of the assembly, so we'll work off the data instead
+			var hash = CryptoUtil.SHA1Hash(File.ReadAllBytes(resolvedPath));
 
-				if (!ResolvedAssemblies.TryGetValue(hash, out var assembly))
+			if (!ResolvedAssemblies.TryGetValue(hash, out var assembly))
+			{
+				assembly = Assembly.LoadFile(resolvedPath);
+				ResolvedAssemblies.Add(hash, assembly);
+
+				// Allow mods to use libraries.
+				var assemblyPath = Path.GetDirectoryName(resolvedPath);
+				if (assemblyPath != null)
 				{
-					assembly = Assembly.LoadFile(resolvedPath);
-					ResolvedAssemblies.Add(hash, assembly);
+					foreach (var referencedAssembly in assembly.GetReferencedAssemblies())
+					{
+						var depedencyPath = Path.Combine(assemblyPath, referencedAssembly.Name + ".dll");
+						if (File.Exists(depedencyPath))
+							LoadAssembly(assemblyList, depedencyPath);
+					}
```

Comment:
We could prolly remove this now when we have the `AssemblyLoader`

Reply:
This would be a nice cleanup, but lets defer it to a followup PR so we can get this one finally merged.